src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case ('addProperties'):
```

Comment:
```suggestion
      case 'addProperties':
```
No need to use brackets for cases